src/Features/CSharp/Portable/CodeRefactorings/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,246 @@
+                            // expression, so that they would be evaluated entirely after "expr" e.g. "expr && (cond1 && cond2)".
+                            // Afterwards, we can directly use it in a case guard e.g. "case pat when cond1 && cond2:".
+                            var condition = (leftmost.Parent.Parent as BinaryExpressionSyntax)
+                                ?.WithLeft(((BinaryExpressionSyntax)leftmost.Parent).Right);
+
+                            return new Pattern.Guarded((Pattern)pattern, (condition ?? node.Right).WalkDownParentheses());
+                        }
+
+                    default:
+                        return null;
+                }
+            }
+
+            protected override bool CanConvertIfToSwitch(IfStatementSyntax ifStatement)
+                => !_semanticModel.AnalyzeControlFlow(ifStatement).ExitPoints.Any(n => n.IsKind(SyntaxKind.BreakStatement));
+
+            protected override IEnumerable<ExpressionSyntax> GetLogicalOrExpressionOperands(
+                ExpressionSyntax syntaxNode)
+            {
+                syntaxNode = syntaxNode.WalkDownParentheses();
+                while (syntaxNode.IsKind(SyntaxKind.LogicalOrExpression))
+                {
+                    var binaryExpression = (BinaryExpressionSyntax)syntaxNode;
+                    yield return binaryExpression.Right.WalkDownParentheses();
+                    syntaxNode = binaryExpression.Left.WalkDownParentheses();
+                }
+
+                yield return syntaxNode;
+            }
+
+            protected override IEnumerable<(StatementSyntax, ExpressionSyntax)> GetIfElseStatementChain(IfStatementSyntax currentStatement)
+            {
+                StatementSyntax elseBody;
+                do
+                {
+                    yield return (currentStatement.Statement, currentStatement.Condition);
+                    elseBody = currentStatement.Else?.Statement;
+
+                    var elseIfStatement = elseBody as IfStatementSyntax;
+                    if (elseIfStatement == null
+                        && !_semanticModel.AnalyzeControlFlow(currentStatement.Statement).EndPointIsReachable
+                        && currentStatement.GetNextStatement() is IfStatementSyntax nextStatement
+                        && CanConvertIfToSwitch(nextStatement))
+                    {
+                        _numberOfSubsequentIfStatementsToRemove++;
+                        currentStatement = nextStatement;
+                    }
+                    else
+                    {
+                        currentStatement = elseIfStatement;
+                    }
+                }
+                while (currentStatement != null);
+
+                yield return (elseBody, null);
+            }
+
+            protected override IEnumerable<SyntaxNode> GetSubsequentIfStatements(IfStatementSyntax ifStatement)
+            {
+                StatementSyntax currentStatement = ifStatement;
```

Comment:
i would prefer 'var'.  so can you just cast the ifStatement to StatementSyntax instead?